SEACAS: Link errors with Nalu after SEACAS update on Mac

CC: @spdomin, @gsjaardema 

## Description

Moving this issue reported in https://github.com/trilinos/Trilinos/issues/13839#issuecomment-2733692252 here.

SHA1: 12808ede9ed1985dfe03449af4ca49bec97b1834

The error actually looks like a STK_IO/Seacas error.. @gsjaardema  Perhaps the lack of testing these past weeks allowed something to sneak through. Is 1.18.12 too old now?

```
Undefined symbols for architecture x86_64:
  "_NC4_hdf5get_libversion", referenced from:
      _ex_config in libexodus.a(ex_utils.c.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [packages/seacas/libraries/ioss/src/main/skinner] Error 1
make[1]: *** [packages/seacas/libraries/ioss/src/main/CMakeFiles/skinner.dir/all] Error 2
```

Config script and configuration attached:

[3_18_2025_MacConfig.txt](https://github.com/user-attachments/files/19322419/3_18_2025_MacConfig.txt)

[do-configTrilinos_nightly_release.txt](https://github.com/user-attachments/files/19322440/do-configTrilinos_nightly_release.txt)

--- a/packages/seacas/libraries/exodus/src/ex_utils.c
+++ b/packages/seacas/libraries/exodus/src/ex_utils.c
@@ -56,7 +56,7 @@ static char *cur_string = &ret_string[0];
 #endif
 
 #ifndef WINDOWS__
-#if NC_HAS_HDF5
+#if (NC_VERSION_MAJOR > 4 || (NC_VERSION_MAJOR == 4 && NC_VERSION_MINOR > 4)) && NC_HAS_HDF5
 extern int NC4_hdf5get_libversion(unsigned *, unsigned *, unsigned *);
 #endif
 #endif
@@ -104,10 +104,12 @@ const char *ex_config(void)
 #ifndef WINDOWS__
 #if NC_HAS_HDF5
   {
+#if NC_VERSION_MAJOR > 4 || (NC_VERSION_MAJOR == 4 && NC_VERSION_MINOR > 4)
     unsigned major, minor, release;
     NC4_hdf5get_libversion(&major, &minor, &release);
     j += snprintf(buffer + j, buffer_size - j, "\t\tHDF5 enabled (%u.%u.%u)\n", major, minor,
                   release);
+#endif
   }
   j += snprintf(buffer + j, buffer_size - j, "\t\tZlib Compression (read/write) enabled\n");
 #if NC_HAS_SZIP_WRITE == 1

